Monika After Story/game/zz_weather.rpy

```diff
@@ -99,16 +99,59 @@ image mas_island_night = "mod_assets/location/special/night_without_frame.png"
 default persistent._mas_weather_MWdata = {}
 # stores locked/unlocked status for weather
 
+default persistent._mas_force_weather = False
+
+#When did we last check if it could rain
+default persistent._mas_date_last_checked_rain = None
+
+#Should it rain today?
+default persistent._mas_should_rain_today = None
+
+#Loading at init 0 because of season functions
+init python in mas_weather:
+    def shouldRainToday():
+        import random
+        import datetime
+
+        #Is it a new day? If so, we should see if it should rain today
+        if not store.persistent._mas_date_last_checked_rain or store.persistent._mas_date_last_checked_rain < datetime.date.today():
+            store.persistent._mas_date_last_checked_rain = datetime.date.today()
+
+            #Now we roll
+            chance = random.randint(0,100)
+
+            #ODDS:
+            #   Spring:
+            #       - 30% chance for it to not rain on a particular day
+            #   Summer:
+            #       - 85% chance for it to not rain on a particular day
+            #   Fall:
+            #       - 40% chance for it to not rain on a particular day
+            if store.mas_isSpring():
+                store.persistent._mas_should_rain_today = chance >= 30
+            elif store.mas_isSummer():
+                store.persistent._mas_should_rain_today = chance >= 85
+            elif store.mas_isFall():
+                store.persistent._mas_should_rain_today = chance >= 40
```

Comment:
should have the else case set the should rain to False. Otherwise it seems that we would always return the last known should rain chance.